### PR TITLE
Ensure Extractor scheduler is requested with non-Zero number of threads

### PR DESCRIPTION
**Pull request by mloskot**
By default, `ExtractorConfig::requested_num_threads` is Zero, which is an invalid value for tbb task scheduler.
If user does not specify value greater than Zero, then use `tbb::task_scheduler_init::automatic`.

### CODE REVIEW

File: src/extractor/extractor.cpp [hunk 1]
```diff
@@ -118,7 +118,7 @@ int Extractor::run(ScriptingEnvironment &scripting_environment)
     TIMER_START(extracting);
 
     const unsigned recommended_num_threads = tbb::task_scheduler_init::default_num_threads();
-    const auto number_of_threads = std::min(recommended_num_threads, config.requested_num_threads);
+    const auto number_of_threads = std::min(recommended_num_threads, std::max(config.requested_num_threads, 1u));
```
Comment (daniel-j-h) [inline src/extractor/extractor.cpp]: Did you check all tools, I think we're doing it in most if not all of them

Comment (mloskot) [inline src/extractor/extractor.cpp]: Nope, I didn't.

One more thing I noticed was that `Extractor::run` initialises the `tbb::task_scheduler_init` based on the config, while `Contractor::Run` does not (seems like leaving it as user's responsibility). I'm unsure, but looks like overlooked interface unification or perhaps by-design. I know too little of the OSRM, didn't want to draw too far-reaching conclusions :)

**Comment by mloskot** [comment]
I've switched from `1u` to `tbb::task_scheduler_init::automatic`.

File: src/extractor/extractor.cpp (hunk 1 above)
Comment (daniel-j-h) [inline src/extractor/extractor.cpp]: Yes probably overlooked or for some other historic reasons.

As I said, all the tools and the library interfaces should be consistent when possible.

Comment (mloskot) [inline src/extractor/extractor.cpp]: > As I said, all the tools and the library interfaces should be consistent when possible.

What does it mean for this PR? Rejection or Need Work? If the latter, what changes you suggest?

Comment (daniel-j-h) [inline src/extractor/extractor.cpp]: No this PR is fine, it just would be great to consolidate the initialization across all tools and interfaces (extract, contract, ..)

Comment (mloskot) [inline src/extractor/extractor.cpp]: Right. Looks like a job for a `{Extractor|Contractor|*Config` factory.

**Pull request merged**